src/ui/base-page-storyEdit.js

```diff
@@ -384,55 +534,74 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/main/LICENSE.
+     *
+     * @return {jqXHR} - the jqXHR for the server request
      */
-    sjrk.storyTelling.base.page.storyEdit.submitStory = function (storyEditorForm, storyModel, errorEvent, storyAutosaveKey) {
-        storyEditorForm.attr({
-            action: "/stories/",
-            method: "post",
-            enctype: "multipart/form-data"
-        });
-
-        // This is the easiest way to be able to submit form
-        // content in the background via ajax
-        var formData = new FormData(storyEditorForm[0]);
-
-        // Stores the entire model as a JSON string in one
-        // field of the multipart form
-        var modelAsJSON = JSON.stringify(storyModel);
-        formData.append("model", modelAsJSON);
-
-        // In the real implementation, this should have
-        // proper handling of feedback on success / failure,
-        // but currently it just logs to console
-        $.ajax({
-            url         : storyEditorForm.attr("action"),
-            data        : formData || storyEditorForm.serialize(),
+    sjrk.storyTelling.base.page.storyEdit.updateStoryOnServer = function (storySaveUrl, storyModel) {
+        return $.ajax({
+            url         : storySaveUrl,
+            data        : JSON.stringify(storyModel),
             cache       : false,
-            contentType : false,
+            contentType : "application/json",
             processData : false,
-            type        : "POST",
-            success     : function (data, textStatus, jqXHR) {
-                // clear the saved story only once it's successfully published
-                sjrk.storyTelling.base.page.storyEdit.clearAutosave(storyAutosaveKey);
-
-                fluid.log(jqXHR, textStatus);
-                var successResponse = JSON.parse(data);
-                var storyUrl = "/storyView.html?id=" + successResponse.id;
-                window.location.assign(storyUrl);
-            },
-            error       : function (jqXHR, textStatus, errorThrown) {
-                fluid.log("Something went wrong");
-                fluid.log(jqXHR, textStatus, errorThrown);
+            type        : "POST"
+        });
+    };
 
-                errorEvent.fire(fluid.get(jqXHR, ["responseJSON", "message"]) || "Internal server error");
-            }
+    /**
+     * @callback publishSuccessCallback - a callback to call after successful publishing
+     * @param {String} storyId - the ID of the story to which the user will be redirected
+     * @param {String} viewPageUrl - the URL for the story View page
+     */
+
+    /**
+     * Sets a story to "published" and redirects the user to the new story page
+     *
+     * @param {Component} storyEditPage - an instance of `sjrk.storyTelling.base.page.storyEdit`
+     * @param {String} viewPageUrl - the URL for the story View page
+     * @param {Component} story - an instance of `sjrk.storyTelling.story`
+     * @param {publishSuccessCallback} successCallback - to call on successful publish
```

Comment:
I wonder if this should just return a promise instead?

Reply:
I think, given that it's called from an invoker which is in turn called in an event listener, I'm not sure how a return value could be accessed.

To stay consistent with the rest of the project, it might make more sense to pass in an event to fire which can return a promise, and then a listener to that event can go from there.

Reply:
I've opted to add an `onStoryPublishSuccess` event that returns the story ID, which allowed me to simplify the function a bit and use listeners to do the rest. I think it's worth revisiting the idea of how the rest of the system could be adapted if `publishStory` returns a promise, though I'm not sure at this point what that would look like